app/views/admin/AdministerUserModal.coffee

```diff
@@ -247,4 +260,175 @@ module.exports = class AdministerUserModal extends ModalView
+
+      if administrated
+        $.ajax
+          type: 'GET',
+          url: '/db/user'
+          data: {
+            fetchByIds: administrated 
+          }
+          success: (teachers) => 
+            @administratedTeachers = teachers or []
+            @updateAdministratedTeachers()
+          error: (jqxhr, status, error) =>
+            errorString = "There was an error getting existing administratedTeachers, see the console"
+            @userSaveState = errorString
+            @render()
+            console.warn errorString, error
+
+    @editingSchoolAdmins = !@editingSchoolAdmins
+    @render()
+
+  onClickAddAdministeredTeacher: (e) -> 
+    teacher = $(e.target).closest('tr').data('user-id')
+    @foundTeachers = @foundTeachers.filter (t) -> t._id isnt teacher
+    @render()
+
+    fetchJson("/db/user/#{@user.id}/schoolAdministrator/administratedTeacher", {
+      method: 'POST',
+      json: {
+        administratedTeacherId: teacher
+      }
+    }).then (res) =>
+      $.ajax
+        type: 'GET',
+        url: '/db/user'
+        data: {
+          adminSearch: teacher
+        }
+        success: (teacher) ->
+          @administratedTeachers.push(teacher)
+          @updateAdministratedTeachers()
+        error: (jqxhr, status, error) =>
+          errorString = "There was an error getting the new administrated teacher, see the console"
+          @userSaveState = errorString
+          @render()
+          console.warn errorString, error
+    null
+
+  onClickRemoveAdministeredTeacher: (e) -> 
+    teacher = $(e.target).closest('tr').data('user-id')
+    @userSaveState = 'removing...'
+
+    @render()
+
+    fetchJson("/db/user/#{@user.id}/schoolAdministrator/administratedTeacher", {
+      method: 'DELETE',
+      json: {
+        administratedTeacherId: teacher
+      }
+    }).then (res) =>
+      @administratedTeachers = @administratedTeacher.filter (t) -> t._id isnt teacher
+      @updateAdministratedTeachers()
+      @userSaveState = null
+      @render()
+    null
+
+  onSearchRequestSuccess: (teachers) =>
+    forms.enableSubmit(@$('#teacher-search-button'))
+    existingTeachers = _.pluck(@administratedTeachers, '_id')
+    existingTeachers.push(@user._id)
+    teachers = _.filter(teachers, (teacher) -> teacher._id not in existingTeachers)
+    result = _.map(teachers, (teacher) ->
+      "
+        <tr data-user-id='#{teacher._id}'>
+          <td>
+            <button class='add-administer-teacher'>Add</button>
+          </td>
+          <td><code>#{teacher._id}</code></td>
+          <td>#{_.escape(teacher.name or 'Anonymous')}</td>
+          <td>#{_.escape(teacher.email)}</td>
+          <td>#{teacher.firstName or 'No first name'}</td>
+          <td>#{teacher.lastName or 'No last name'}</td>
+          <td>#{teacher.schoolName or 'No school name'}</td>
+          <td>Verified teacher: #{teacher.verifiedTeacher or 'false'}</td>
+        </tr>
+      "
```

Comment:
Should we building HTML this way?  It seems like we should have data driving a template somewhere instead.

Reply:
Yes I agree. This was inspired from the existing search box on the admin page. Let me make another pass and structure this more from data filling the template.